app/vmalert/utils/metrics.go

```diff
@@ -18,10 +20,10 @@ type Gauge struct {
 }
 
 // GetOrCreateGauge creates a new Gauge with the given name
-func GetOrCreateGauge(name string, f func() float64) *Gauge {
+func GetOrCreateGauge(set *metrics.Set, name string, f func() float64) *Gauge {
```

Comment:
I think we should change all the `GetOrCreatexx` to `Createxx`, since it must be a bug if vmalert panics due to metric name duplication, and it helps catch the bug.

Reply:
Good point, updated